### PR TITLE
Implemented max-width for team-, affiliate-names

### LINKED ISSUE #431: Truncate team name on scoreboard
Currently a long team name get displayed in full length on the scoreboard. This creates a suboptimal layout wasting a lot of space on the team column and potentially pushing the problem columns out of the window. We should probably restrict the team column to a maximum width, and truncate the team name (don't wrap to a new line, since it would create rows with uneven height).
Attached a screenshot from UKIEP. Note also that the category leader button adds to the problem.
screen

### PR DESCRIPTION
**Pull request by tuupke**
Added a configuration option to set the max-width for the affiliation and team names. When set to 0, it does not apply a maximum width. Some JS adds a class when the contents are too wide, which adds a slot that houses the gradient, this gradient overlaps with the rightmost 30% of the field. It has been tested in Webkit browsers (opera, safari) and Firefox

Old:
old

New:
new

Fixes #431

### CODE REVIEW

File: sql/mysql_db_defaultdata.sql [hunk 1]
```diff
@@ -59,7 +59,8 @@ INSERT INTO `configuration` (`name`, `value`, `type`, `public`, `category`, `des
 ('openid_clientid', '""', 'string', '0','Misc', 'OpenID Connect client id'),
 ('openid_clientsecret', '""', 'string', '0', 'Misc', 'OpenID Connect client secret'),
 ('data_source', '0', 'int', '0', 'Misc', 'Source of data. Choices: 0 = all local, 1 = configuration data external, 2 = configuration and live data external'),
-('ip_autologin', '0', 'bool', '0', 'Misc', 'Enable to skip the login page when using IP authentication.');
+('ip_autologin', '0', 'bool', '0', 'Misc', 'Enable to skip the login page when using IP authentication.'),
+('team_column_width', '0', 'int', '0', 'Misc', 'Maximum width of team column on scoreboard. Leave 0 for no maximum.');
```
Comment (nickygerritsen) [inline sql/mysql_db_defaultdata.sql]: How about putting it in display options?

File: webapp/src/DOMJudgeBundle/Twig/TwigExtension.php [hunk 2]
```diff
@@ -893,4 +894,32 @@ public function wrapUnquoted(string $text, int $width = 75, string $quote = '>')
...
+    public function hexColorToRGBA(string $text, float $opacity = 1): string
+    {
+        $col = Utils::convertToHex($text);
+        preg_match_all("/[0-9A-Fa-f]{2}/", $col,$m);
+        if (!count($m)) {
+            return $text;
+        }
+
+        $m = current($m);
+        switch (count($m)){
```
Comment (nickygerritsen) [inline webapp/src/DOMJudgeBundle/Twig/TwigExtension.php]: Some formatting here :)

File: webapp/src/DOMJudgeBundle/Resources/views/partials/scoreboard_table.html.twig [hunk 3]
```diff
@@ -323,3 +337,49 @@
         </table>
     {% endif %}
 {% endif %}
+
+<style>
+    {% for color,i in backgroundColors %}
+    {% set cssColor = "#"~color %}
```
Comment (nickygerritsen) [inline webapp/src/DOMJudgeBundle/Resources/views/partials/scoreboard_table.html.twig]: Should we indent this one more time?

**Comment by nickygerritsen** [comment]
Looks good to me in general, some small comments

File: sql/mysql_db_defaultdata.sql (hunk 1 above)
Comment (tuupke) [reply sql/mysql_db_defaultdata.sql]: Changed in 2a5408eb58df7c7126c411d447e4a3578d0cc0d8

File: webapp/src/DOMJudgeBundle/Twig/TwigExtension.php (hunk 2 above)
Comment (tuupke) [reply webapp/src/DOMJudgeBundle/Twig/TwigExtension.php]: Resolved in f7ccd52cd87e43d97000bd0aca80af5607a28160

File: webapp/src/DOMJudgeBundle/Resources/views/partials/scoreboard_table.html.twig (hunk 3 above)
Comment (tuupke) [reply webapp/src/DOMJudgeBundle/Resources/views/partials/scoreboard_table.html.twig]: Resolved in f7ccd52cd87e43d97000bd0aca80af5607a28160

**Comment by meisterT** [comment]
Does it show the original name if you hover over it?

**Pull request merged**